util/buildflags/cache.go

```diff
@@ -79,3 +80,24 @@ func addGithubToken(ci *client.CacheOptionsEntry) bool {
 	}
 	return ci.Attrs["token"] != "" && ci.Attrs["url"] != ""
 }
+
+func addAwsCredentials(ci *client.CacheOptionsEntry) {
+	if ci.Type != "s3" {
+		return
+	}
+	if _, ok := ci.Attrs["access_key_id"]; !ok {
```

Comment:
This should be done automatically by the AWS SDK. Are you sure we really need to do this?

Reply:
I mean we the SDK should be able to configure itself without any manual mapping, For example it's working with AWS IAM Profile. We should look on how to do the same with these env variables.

Reply:
Spent some time to think about it.
Your solution will work only for simple authent (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`). The SDK offers plenty of solutions to authent (example by using `AWS_PROFILE`), but I do not think there is a simple way to authenticate inside the client and forward creds to the server. Or it do not worth it. So dealing only with simple authent is fine for me, if it's clear in the doc.